### PR TITLE
fix(vm): fixed startup / shutdown behaviour on HA clusters

### LINKED ISSUE #507: Changing a disk's datastore when HA is enabled causes various problems
**Describe the bug**

I have some VMs that have been created on the Proxmox cluster and added as HA resources. I am trying to modify the datastore for some disks on these VMs.

When trying to do that, the plugin gets confused by the fact that sending a shutdown command to a HA-enabled VM returns immediately, without the VM having been shut down. Because of this fact:

  * the plugin moves on to trying to move the disk; 
  * the Proxmox cluster's HA manager tries to shutdown the VM, which is now locked by the disk move operation;
  * after a few attempts, the Proxmox cluster sets the VM's HA state to `error`.

On some of the VMs the disk was not fully moved (copy created on new datastore but disk hardware unchanged), on others there is a residual `Unused disk` for which no Rados image actually exists.

**To Reproduce**
1. Create a Terraform resource for a few VMs with one disk on some datastore,
2. Manually add the VMs as HA resources,
3. Update the Terraform resource in order to change the disks' datastore.
4. Run Terraform
5. Watch the fireworks happening on the Proxmox cluster

**Expected behavior**
The VMs should shut down, then the disks should be moved, and then they should restart.

**Screenshot**

lock-failure

The third line here is unrelated, but the other lines clearly show the problem.

**Additional context**

- The issue doesn't occur systematically. Trying to do it on multiple VMs in parallel improves the chances of triggering it.
- Some of the additional mess with the disk's configuration may be my fault as I tried to set the HA state to `disabled` while the operation was still running.
- Using latest version + my own HA support branch, but I also tested with a manually-managed HA resource.
- My datastores are all Ceph-based, although I don't believe this has any influence.

### PR DESCRIPTION
**Pull request by tseeker**
When a Proxmox cluster receives a shutdown or startup command for a VM that is declared as a HA resource, it doesn't honour the request immediately. Instead the API call returns and the state changes later, at the HA manager's discretion. Because of that, the plugin could end up trying to effect inappropriate changes on a VM that wasn't in the expected state.

This PR adds a wait loop after both startup and shutdown commands in order to ensure that the state is what we expect it to be.

### Contributor's Note
Please mark the following items with an [x] if they apply to your PR.
Leave the [ ] if they are not applicable, or if you have not completed the item.
- [ ] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [ ] I have added / updated templates in `/examples` for any new or updated resources / data sources.
- [x] I have ran `make examples` to verify that the change works as expected. 

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #507

### CODE REVIEW

**Comment by bpg** [comment]
@all-contributors please add @tseeker for bug, tests

File: proxmoxtf/resource/vm.go [hunk 1]
```diff
@@ -1600,7 +1600,44 @@ func deleteIdeDrives(ctx context.Context, vmAPI *vms.Client, itf1 string, itf2 s
 	return nil
 }
 
-// Shutdown the VM.
+// Wait until a VM reaches the desired state. An error is returned if the VM doesn't reach the
+// desired state within the given timeout.
+func vmWaitState(ctx context.Context, vmAPI *vms.Client, state string, timeout int) diag.Diagnostics {
```
Comment (bpg-dev) [inline proxmoxtf/resource/vm.go]: We can probably reuse this one? https://github.com/bpg/terraform-provider-proxmox/blob/9fd9d211d75e760ef1c7e44d13de9ce8d38bf834/proxmox/nodes/vms/vms.go#L549

**Review by bpg — APPROVED** [review]
Updated to use the existing function as per my comment above. 

LGTM! 🚀

File: proxmoxtf/resource/vm.go (hunk 1 above)
Comment (tseeker) [reply proxmoxtf/resource/vm.go]: Oh! I hadn't seen it. Thanks for catching that.

**Pull request merged**